src/main/flight/mixer.c

```diff
@@ -346,14 +346,92 @@ static void applyFlipOverAfterCrashModeToMotors(void)
     }
 }
 
+#ifdef USE_RPM_LIMITER
+static void applyRPMLimiter(void)
+{
+    if (mixerConfig()->rpm_limiter && motorConfig()->dev.useDshotTelemetry) {
+        float scaledRPMLimit = 0;
+        float pidOutput = 0;
+        float rcCommandThrottle = (rcCommand[THROTTLE] - 1000) / 1000.0f;
+        float averageRPM = 0;
+        float averageRPMSmoothed = 0;
+        bool motorsSaturated = false;
+        for (int i = 0; i < getMotorCount(); i++) {
+            averageRPM += getDshotTelemetry(i);
+            if (motor[i] >= motorConfig()->maxthrottle) {
+                motorsSaturated = true;
+            }
+        }
+        averageRPM = 100 * averageRPM / (getMotorCount() * motorConfig()->motorPoleCount / 2.0f);
+        averageRPMSmoothed = mixerRuntime.rpmLimiterPreviousSmoothedRPM + mixerRuntime.rpmLimiterDelayK * (averageRPM - mixerRuntime.rpmLimiterPreviousSmoothedRPM); //kinda braindead to convert to rps then back
+        if (mixerConfig()->rpm_limiter_rpm_linearization) {
+            scaledRPMLimit = ((mixerConfig()->rpm_limiter_rpm_limit - mixerConfig()->rpm_limiter_idle_rpm)) * 100.0f * (rcCommandThrottle) + mixerConfig()->rpm_limiter_idle_rpm * 100.0f;
```

Comment:
based on this line of code, a check needs to be made to ensure that rpm_limiter_idle_rpm cannot be bigger than the rpm_limiter_rpm_limit